types/src/jwks/rsa.rs

```diff
@@ -33,6 +37,20 @@ impl RSA_JWK {
         }
     }
 
+    pub fn to_poseidon_scalar(&self) -> Result<ark_bn254::Fr> {
+        let mut modulus = base64::decode_config(&self.n, URL_SAFE_NO_PAD)?;
+        modulus.reverse(); // This is done to match the circuit, which requires the modulus in a verify specific format due to how RSA verification is implemented
+                           // TODO(zkid): finalize the jwk hashing scheme.
+        let mut scalars = modulus
+            .chunks(24) // Pack 3 64 bit limbs per scalar, so chunk into 24 bytes per scalar
+            .map(|chunk| {
+                poseidon_bn254::pack_bytes_to_one_scalar(chunk).expect("chunk converts to scalar")
+            })
+            .collect::<Vec<ark_bn254::Fr>>();
+        scalars.push(ark_bn254::Fr::from(RSA_MODULUS_BYTES as i32));
```

Comment:
`RSA_MODULUS_BYTES` is 256. What if self is an RS384 key?
Currently if a provider returns a RSA384 key, jwk consensus can support it.

Reply:
JWK consensus should definitely support it, for the zkless path to work, even though the ZK path will not support it.

@heliuchuan, you are best familiar with the code, so we just need to make sure that the ZK path properly rejects Groth16-based zkID signatures whose JWK is **not** an RSA 256-byte key.

Reply:
Done, added check for 256 length key.